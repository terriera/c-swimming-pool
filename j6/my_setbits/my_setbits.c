/*
** my_setbits.c for 2002 swimming pool
**
** Made by gumhead
**
*/

unsigned int	my_setbits(unsigned nb, int n, int p, unsigned x)
{
  unsigned int	mask;

  mask = ~(~0 << p);
  return ((x & mask) << n) | (nb & ~(mask << n));
}
