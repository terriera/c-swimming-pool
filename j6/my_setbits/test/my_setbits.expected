00000000000000000000000001010000
11111111111111111111011100011111
11111010001010110011110001111111
